aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -86,33 +86,50 @@ private AWSDataStorePlugin(
     }
 
     /**
-     * Creates an {@link AWSDataStorePlugin} which can warehouse the model types provided by
+     * Constructs an {@link AWSDataStorePlugin} which can warehouse the model types provided by
+     * your application's code-generated model provider. This model provider is expected to have the
+     * fully qualified class name of com.amplifyframework.datastore.generated.model.AmplifyModelProvider.
+     *
+     * This constructor will attempt to find that provide by means of reflection. If you have changed
+     * the path to your {@link ModelProvider} instance, and do not use the default path generated by
+     * the Amplify Code Generator, then this will be break. Likewise, if you need to provide a custom
+     * {@link ModelProvider}, this will not work. In both cases, you should prefer one of the other
+     * overloads such as {@link #AWSDataStorePlugin(ModelProvider)}.
+     *
+     * If remote synchronization is enabled, it will be performed via {@link Amplify#API}.
+     *
+     * @throws DataStoreException If it is not possible to access the code-generated model provider
+     */
+    @SuppressWarnings({"unused", "WeakerAccess"})
+    public AWSDataStorePlugin() throws DataStoreException {
+        this(ModelProviderLocator.locate(), Amplify.API);
```

Comment:
oh wow, this is fancy! I like it.

I wasn't expecting this to happen anytime soon, you're a ninja @jamesonwilliams